src/main/java/net/kaczmarzyk/spring/data/jpa/domain/PathSpecification.java

```diff
@@ -56,6 +56,16 @@ public PathSpecification(QueryContext queryContext, String path) {
         return (Path<F>) expr;
     }
 
+	private Path<T> getEvaluatedPath(String field, Root<T> root) {
+		Path<T> evaluated = (Path<T>) queryContext.getEvaluated(field, root);
+
+		if(evaluated == null) {
```

Comment:
1. add a short comment on when it can happen that evaluated is null (maybe some quick example of the annotation config?
2. Maybe `queryContext.getEvaluated` could contain this `if null then getEvaluatedJoinFetch` ?